output/full.go

```diff
@@ -40,6 +40,12 @@ func submitFull(ctx context.Context, s snapshot.FullSnapshot, server *state.Serv
 	var err error
 	var data []byte
 
+	err = verifyIntegrity(s)
+	if err != nil {
+		logger.PrintError(fmt.Sprintf("Snapshot integrity check failed: %s; please contact support", err))
+		return err
```

Comment:
I wonder if there is an argument to be made that this should not return an error - because as I noticed when working on #401, failing the snapshot run causes the state update to be discarded (since the [relevant code portion is not run](https://github.com/pganalyze/collector/blob/main/runner/full.go#L233)) - and thus the next full snapshot would use the previous reference point (potentially running into the exact same integrity problem again, if its related to the diffing to the prior data point).

If we change this to not return an error, we should consider whether we'd still want to submit a failed snapshot (i.e. call `SendFailedFull`), which would otherwise have been done by [CollectAllServers](https://github.com/pganalyze/collector/blob/main/runner/full.go#LL221) when `SendFull` returns an error.

Reply:
Hmm, interesting. I don't have a strong opinion on `SendFailedFull` but it seems like it might be a good idea to include these messages in the UI.

Separately, I'm also wondering if maybe we should allow ignoring the integrity check. For example, negative stats are clearly bogus but could in theory be localized, and it might be inconvenient for all server stats to stop working just because of that... I don't want to just log errors, because I think we should only check things here that are clear bugs, and if they're only logged, we won't hear about them. But we may also want an escape hatch (unless the problem is catastrophic, like the relation / info length mismatches).

Reply:
> Separately, I'm also wondering if maybe we should allow ignoring the integrity check.

Yeah, maybe? An escape hatch seems useful, but I'm also wary of adding too many flags that are not used. Maybe this is a good use case for an extra command line flag? (and not a new config setting)

Reply:
Uhmm, or we could skip negative stats check for now 🤔 I think reference + information length check is good enough as a starter and I don't think we need to allow ignoring these checks.

Reply:
I thought about that, but I don't think omitting checks like that is beneficial: negative numbers there _are_ clearly bogus, even if they're not as problematic for snapshot processing. Still, it does have the benefit of not having to introduce a workaround (yet?), since we clearly don't want to send snapshots with critical integrity issues. Anyone else have thoughts on this?

Reply:
>I wonder if there is an argument to be made that this should not return an error - because as I noticed when working on https://github.com/pganalyze/collector/pull/401, failing the snapshot run causes the state update to be discarded (since the [relevant code portion is not run](https://github.com/pganalyze/collector/blob/main/runner/full.go#L233)) - and thus the next full snapshot would use the previous reference point (potentially running into the exact same integrity problem again, if its related to the diffing to the prior data point).

@lfittl is there a class of bug that prevents us from updating the state after a failed run?

Reply:
I wonder how many negative numbers are currently sent and processed. As you mentioned, they're not problematic for snapshot processing and it will allow the other potentially-good data (e.g. server metrics) to be stored. With this change, such things will be dropped.
I did some check, looks like we had 11 servers having the report of negative `n_tup_ins` within one day.

Reply:
We can always move the stats check to snapshot processing, so only that data is skipped.